test/common/network/lc_trie_speed_test.cc

```diff
@@ -5,136 +5,145 @@
 
 namespace {
 
-std::vector<Envoy::Network::Address::InstanceConstSharedPtr> addresses;
-
-std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>> tag_data;
-
-std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>>
-    tag_data_nested_prefixes;
-
-std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>>
-    tag_data_minimal;
-
-std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> lc_trie;
+struct AddressInputs {
+  AddressInputs() {
+    // Random test addresses from RFC 5737 netblocks
+    static const std::string test_addresses[] = {
+        "192.0.2.225",   "198.51.100.55", "198.51.100.105", "192.0.2.150",   "203.0.113.162",
+        "203.0.113.110", "203.0.113.99",  "198.51.100.23",  "198.51.100.24", "203.0.113.12"};
+    for (const auto& address : test_addresses) {
+      addresses_.push_back(Envoy::Network::Utility::parseInternetAddress(address));
+    }
+  }
 
-std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> lc_trie_nested_prefixes;
+  std::vector<Envoy::Network::Address::InstanceConstSharedPtr> addresses_;
+};
+
+struct CidrInputs {
+  CidrInputs() {
+    // Construct three sets of prefixes: one consisting of 1,024 addresses in an
+    // RFC 5737 netblock, another consisting of those same addresses plus
+    // 0.0.0.0/0 (to exercise the LC Trie's support for nested prefixes),
+    // and finally a set containing only 0.0.0.0/0.
+    for (int i = 0; i < 32; i++) {
+      for (int j = 0; j < 32; j++) {
+        tag_data_.emplace_back(
+            std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>(
+                {"tag_1",
+                 {Envoy::Network::Address::CidrRange::create(
+                     fmt::format("192.0.{}.{}/32", i, j))}}));
+      }
+    }
+    tag_data_nested_prefixes_ = tag_data_;
+    tag_data_nested_prefixes_.emplace_back(
+        std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>(
+            {"tag_0", {Envoy::Network::Address::CidrRange::create("0.0.0.0/0")}}));
+    tag_data_minimal_.emplace_back(
+        std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>(
+            {"tag_1", {Envoy::Network::Address::CidrRange::create("0.0.0.0/0")}}));
+  }
 
-std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> lc_trie_minimal;
+  std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>> tag_data_;
+  std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>>
+      tag_data_nested_prefixes_;
+  std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>>
+      tag_data_minimal_;
+};
 
 } // namespace
 
 namespace Envoy {
 
 static void BM_LcTrieConstruct(benchmark::State& state) {
```

Comment:
I'm not sure how to best address the clang tidy warnings.  BM_ prefixes are the recommended naming convention recommended by benchmark/benchmark.h
I dropped the BM_ prefix.

static void BM_LcTrieConstruct(benchmark::State& state) {
            ^~~~~~~~~~~~~~~~~~
            bmLcTrieConstruct
/build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/test/common/network/lc_trie_speed_test.cc:69:13: error: invalid case style for function 'BM_LcTrieConstructNested' [readability-identifier-naming,-warnings-as-errors]